### DISCUSSION TITLE
Class changes not applying after initial postcss processing

### DISCUSSION THREAD

**Discussion by stevenpslade** [tailwindlabs/tailwindcss#8057] [Help] [answered]
### Problem

I'm using React, Tailwind CSS v3.0, and Gulp/gulp-postcss. If I am using a tailwind class like `text-green-600` and run `npx gulp css`, the color will apply. However, if I change the class to `text-red-600` the new class does not effect until I run the postcss process again or restart my server.

### Packages

 - react: "^17.0.2"
 - tailwindcss: "^3.0.23"
 - gulp-postcss: "9.0.1"
 
### Config Files
 
**tailwind.config.js:**
 ```
 module.exports = {
  darkMode: 'class',
  content: [
    './src/**/*.{js,jsx,ts,tsx}',
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
 ```
 
**postcss.config.js:**
 ```
 module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  }
}
 ```
 
**gulpfile.js:**
Note that according to [gulp-postcss docs](https://github.com/postcss/gulp-postcss#basic-usage), the configuration is loaded automatically from postcss.config.js, so you don't have to specify any options.
 ```
const gulp = require("gulp");
const postcss = require("gulp-postcss");
const sourcemaps = require("gulp-sourcemaps");

gulp.task("css", () => {
  return gulp.src('src/index.css')
    .pipe( sourcemaps.init() )
    .pipe( postcss() )
    .pipe( sourcemaps.write('.') )
    .pipe( gulp.dest('./public') )
});
 ```
 
 ### Other Considerations
 
 I know Tailwind CSS v3.0 is using JIT mode by default. I have seen other issues that discuss needing to upgrade `postcss`. I have checked that the version of `gulp-postcss` in my project is using an up to date version of `postcss`.
 
 The file I'm testing my class changes with is `./src/components/Address.jsx`. This matches the pattern configured in my `tailwind-config.js` file.
 
 Anytime I rerun `npx gulp css` the new class I'm using takes effect.

**Comment by stevenpslade** [comment, marked as answer]
I have potentially misunderstood how Tailwind CSS v3 and JIT work. It appears that I should either be watching the css with something like:
```
npx tailwindcss -i ./src/index.css -o ./public/index.css --watch
```

or using the tailwind cdn in development to always have the classes available. For production this would not be an issue because I would process the css during the build and the classes used in the app would be available.

**Reply by adamwathan** [reply]
Yeah unfortunately the Gulp PostCSS plugin is outdated and doesn't support some of the latest PostCSS features that Tailwind needs. Best bet is to use our CLI like you've suggested for sure if you want things to always work perfectly.

**Reply by stevenpslade** [reply]
💯 Adding the cdn is easy and works great:
Added to the head of **./public/index.html** in my React app:
```
<% if (process.env.NODE_ENV === 'development') { %>
  <script src="https://cdn.tailwindcss.com"></script>
<% } %>
```

**Comment by Serenit8** [comment]
create a custom CSS preprocessor which compiles Sass syntax and supplements it with further features. It’s ideal for anyone with specific CSS requirements who knows a little Node.js.
[PerYourHealth Login](https://www.peryourhealth.biz/)
